app/models/spaceship.rb

```diff
@@ -1,102 +1,113 @@
   def get_unequipped_equipment
-    Item.where(spaceship: self, equipped: false).where("loader LIKE ?", "equipment%")
+    get_equipment.where(equipped: false)
   end
 
   # Get equipped Equipment in Ship
   def get_equipped_equipment
-    Item.where(spaceship: self, equipped: true).where("loader LIKE ?", "equipment%")
+    get_equipment.where(equipped: true)
+  end
+
+  def get_equipped_equipment_of_slot_type(type)
+    type = type.to_s
+    self.get_equipped_equipment.each_with_object([]) do |item, result|
+      result << item if item.get_attribute(:slot_type) == type
+    end
+  end
+
+  def get_equipped_equipment_of_type(type)
+    type = type.to_s
+    self.get_equipped_equipment.each_with_object([]) do |item, result|
+      result << item if item.get_attribute(:type) == type
+    end
   end
 
   # Get Main Equipment in Ship
   def get_main_equipment(active = false)
-    items = []
-    self.get_equipped_equipment.each do |item|
-      active ? (items << item if (item.get_attribute('slot_type') == "main") && item.active) : (items << item if item.get_attribute('slot_type') == "main")
-    end
+    items = get_equipped_equipment_of_slot_type(:main)
+    items = items.select(&:active) if active
     items
   end
 
   # Get Utility Equipment in Ship
   def get_utility_equipment
-    items = []
-    self.get_equipped_equipment.each do |item|
-      items << item if item.get_attribute('slot_type') == "utility"
-    end
-    items
+    get_equipped_equipment_of_slot_type(:utility)
   end
 
   # Deactivate Equipment
   def deactivate_equipment
     self.get_equipped_equipment.each do |item|
-      item.update_columns(active: false) if item.active
+      item.update(active: false)
     end
   end
 
   # Deactivate Weapons
   def deactivate_weapons
-    self.get_equipped_equipment.each do |item|
-      item.update_columns(active: false) if item.active && (item.get_attribute('type') == "Weapon")
+    self.get_equipped_equipment_of_type('Weapon').each do |item|
+      item.update(active: false)
     end
   end
 
   # Deactivate Selfrepair Equipment
   def deactivate_selfrepair_equipment
-    self.get_equipped_equipment.each do |item|
-      item.update_columns(active: false) if item.active && (item.get_attribute('type') == "Repair Bot")
+    self.get_equipped_equipment_of_type('Repair Bot').each do |item|
+      item.update_columns(active: false)
     end
   end
 
   # Deactivate Remoterepair Equipment
   def deactivate_remoterepair_equipment
-    self.get_equipped_equipment.each do |item|
-      item.update_columns(active: false) if item.active && (item.get_attribute('type') == "Repair Beam")
+    self.get_equipped_equipment_of_type('Repair Beam').each do |item|
+      item.update_columns(active: false)
     end
   end
 
   # Drop Loot
   def drop_loot
-    items = self.get_items
-    if items.present?
-      structure = Structure.create(location: self.user.location, structure_type: 'wreck')
+    if self.items.size.positive?
```

Comment:
Can we use `self.items.present?` here instead?

Reply:
correct.   i was just replacing `> 0` with `.positive?` which is a new Rubocop recommendation.  I have no strong feelings about it either way. :)

Reply:
I personally would much prefer present? here instead of positive :) There is also another place in the code where this is used and I think its not so good to read.

Reply:
as you wish